classes/task/update_suppression_list.php

```diff
@@ -0,0 +1,175 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Scheduled task for updating the email suppression list.
+ *
+ * @package    tool_emailutils
+ * @copyright  2024 onwards Catalyst IT {@link http://www.catalyst-eu.net/}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @author     Waleed ul hassan <waleed.hassan@catalyst-eu.net>
+ */
+
+namespace tool_emailutils\task;
+
+defined('MOODLE_INTERNAL') || die();
```

Comment:
You don't need a `MOODLE_INTERNAL` check on a file that just defines a single class, only if there are "side effects" like including other files directly, or defining/echoing things outside of the class.